language/vm/serializer-tests/tests/serializer_tests.rs

```diff
@@ -32,3 +32,22 @@ proptest! {
         prop_assert_eq!(module, deserialized_module);
     }
 }
+
+#[test]
+fn serialize_and_deserialize_deeply_nested_types() {
+    let mut m = empty_module();
+    let mut ty = SignatureToken::U8;
+    // TODO: turn the number of iterations up once we fix Drop for SignatureToken.
+    for _ in 0..10000 {
```

Comment:
nit constant for the `10000` you should be able to declare it inside the function scope